python/ray/train/_internal/stats.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import threading
+from typing import Dict, Optional
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.schema import TrainDatasetInfo, TrainRunInfo, TrainWorkerInfo
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote(num_cpus=0)
+class TrainStatsActor:
+    def __init__(self):
+        self.train_runs = dict()
+
+    def register_train_run(self, run_info: TrainRunInfo):
+        # Register a new train run.
+        self.train_runs[run_info.id] = run_info
+
+    def get_train_run(self, run_id: str) -> Optional[TrainRunInfo]:
+        # Retrieve a registered run with its id
+        return self.train_runs.get(run_id, None)
+
+    def get_all_train_runs(self) -> Dict[str, TrainRunInfo]:
+        # Retrieve all registered train runs
+        return self.train_runs
+
+
+TRAIN_STATS_ACTOR_NAME = "train_stats_actor"
+TRAIN_STATS_ACTOR_NAMESPACE = "_train_stats_actor"
+
+_stats_actor_lock: threading.RLock = threading.RLock()
+
+
+def get_or_launch_stats_actor():
+    """Create or launch a `TrainStatsActor` on the head node."""
+    with _stats_actor_lock:
+        return TrainStatsActor.options(
+            name=TRAIN_STATS_ACTOR_NAME,
+            namespace=TRAIN_STATS_ACTOR_NAMESPACE,
+            get_if_exists=True,
+            lifetime="detached",
+            resources={"node:__internal_head__": 0.001},
```

Comment:
This seems to be a internal resource type that is not frequently used.

The solution proposed [here](https://github.com/ray-project/ray/issues/15930) is for local mode, which gets the node id of head node with `ray.get_runtime_context().node_id`, then launch with `NodeAffinityScheduleingStrategy`. However, we almost always call function on a remote worker so it's impossible to get the head node id.

Reply:
I think we can just pass in `ray.get_runtime_context().node_id` and we'll call it for the first time on the driver (on the head node).

Then, the resources get ignored on any subsequent calls due to `get_if_exists`.

Reply:
Actually this `get_or_launch_stats_actor ` method is called in the train trainable, which could be on a worker node.  Also some advanced users probably launch TorchTrainer in a remote task.

The most straightforward solution is to have an API in ray core, so that we can get the head node ip.